In [1]:
import pandas as pd # pandas
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
from sklearn.impute import SimpleImputer
# from sklearn import impute
from  scipy.stats import skew, kurtosis, shapiro



In [27]:
attributes = pd.read_csv("D:\Onedrive\Leuven\Final project\data\\addinfo.csv",index_col = "meter_id")
print(attributes.shape)
amountMeters = attributes.shape[0]
attributes.head(100)

(2143, 23)


dwelling_type  \
meter_id                                                          
0x02b7418b821b2edeeb2baa23ba1580566b4b0e43  semi_detached_house   
0xaf32b2b497e9eb9c3f3de907aedc6a7914a11f41       terraced_house   
0xdd45ab28a7a40eadc04607d6da42e5ad923b1339  semi_detached_house   
0x2b538a5ebfc3780aef25227267063cb6410d943b       detached_house   
0xee032ed498c9e99a4c65e0f39896fb30e513c6b7                  NaN   
...                                                         ...   
0xd33e5ad8e928a17e08384bea04c0d583edc77c2c       terraced_house   
0xc053c89ca63bad876806ebd23e6cbcdf71b8b1a3                  NaN   
0xcef4d8d3c2bde0565ceac3814cf8ef43a25b4963                  NaN   
0xebc6019ab4ef2300113d585a6fb9e85656342558       terraced_house   
0x3fa8effb7c96a42b8600e5e8eace26cbdde14c46       detached_house   

                                            num_occupants  num_bedrooms  \
meter_id                                                                  
0x02b7418b821b2edeeb2baa23ba1580566b4b0e43            NaN           3.0   
0xaf32b2b497e9eb9c3f3de907aedc6a7914a11f41            NaN           2.0   
0xdd45ab28a7a40eadc04607d6da42e5ad923b1339            NaN           NaN   
0x2b538a5ebfc3780aef25227267063cb6410d943b            NaN           4.0   
0xee032ed498c9e99a4c65e0f39896fb30e513c6b7            NaN           NaN   
...                                                   ...           ...   
0xd33e5ad8e928a17e08384bea04c0d583edc77c2c            NaN           3.0   
0xc053c89ca63bad876806ebd23e6cbcdf71b8b1a3            NaN           NaN   
0xcef4d8d3c2bde0565ceac3814cf8ef43a25b4963            NaN           2.0   
0xebc6019ab4ef2300113d585a6fb9e85656342558            NaN           3.0   
0x3fa8effb7c96a42b8600e5e8eace26cbdde14c46            NaN           5.0   

                                           heating_fuel hot_water_fuel  \
meter_id                                                                 
0x02b7418b821b2edeeb2baa23ba1580566b4b0e43          NaN            NaN   
0xaf32b2b497e9eb9c3f3de907aedc6a7914a11f41          NaN            NaN   
0xdd45ab28a7a40eadc04607d6da42e5ad923b1339          NaN            NaN   
0x2b538a5ebfc3780aef25227267063cb6410d943b          NaN            NaN   
0xee032ed498c9e99a4c65e0f39896fb30e513c6b7          NaN            NaN   
...                                                 ...            ...   
0xd33e5ad8e928a17e08384bea04c0d583edc77c2c          NaN            NaN   
0xc053c89ca63bad876806ebd23e6cbcdf71b8b1a3          NaN            NaN   
0xcef4d8d3c2bde0565ceac3814cf8ef43a25b4963          NaN            NaN   
0xebc6019ab4ef2300113d585a6fb9e85656342558          NaN            NaN   
0x3fa8effb7c96a42b8600e5e8eace26cbdde14c46          NaN            NaN   

                                           boiler_age loft_insulation  \
meter_id                                                                
0x02b7418b821b2edeeb2baa23ba1580566b4b0e43        NaN             NaN   
0xaf32b2b497e9eb9c3f3de907aedc6a7914a11f41        NaN             NaN   
0xdd45ab28a7a40eadc04607d6da42e5ad923b1339        NaN             NaN   
0x2b538a5ebfc3780aef25227267063cb6410d943b        NaN             NaN   
0xee032ed498c9e99a4c65e0f39896fb30e513c6b7        NaN             NaN   
...                                               ...             ...   
0xd33e5ad8e928a17e08384bea04c0d583edc77c2c        NaN             NaN   
0xc053c89ca63bad876806ebd23e6cbcdf71b8b1a3        NaN             NaN   
0xcef4d8d3c2bde0565ceac3814cf8ef43a25b4963        NaN             NaN   
0xebc6019ab4ef2300113d585a6fb9e85656342558        NaN             NaN   
0x3fa8effb7c96a42b8600e5e8eace26cbdde14c46        NaN             NaN   

                                           wall_insulation  \
meter_id                                                     
0x02b7418b821b2edeeb2baa23ba1580566b4b0e43             NaN   
0xaf32b2b497e9eb9c3f3de907aedc6a7914a11f41             NaN   
0xdd45ab28a7a40eadc04607d6da42e5ad923b133

There are 2143 meters with additional information of at least one of the 23 attributes.

In [24]:
# It is possible to look to the month December where all series have data to investigate the effect of an attribute. 
for col in attributes.columns:
    total = amountMeters - attributes[col].isnull().sum()
    print("Attribute %s has %s filled places." % (col, total))


Attribute dwelling_type has 1702 filled places.
Attribute num_occupants has 74 filled places.
Attribute num_bedrooms has 1859 filled places.
Attribute heating_fuel has 78 filled places.
Attribute hot_water_fuel has 76 filled places.
Attribute boiler_age has 74 filled places.
Attribute loft_insulation has 75 filled places.
Attribute wall_insulation has 75 filled places.
Attribute heating_temperature has 74 filled places.
Attribute efficient_lighting_percentage has 73 filled places.
Attribute dishwasher has 76 filled places.
Attribute freezer has 70 filled places.
Attribute fridge_freezer has 70 filled places.
Attribute refrigerator has 73 filled places.
Attribute tumble_dryer has 76 filled places.
Attribute washing_machine has 76 filled places.
Attribute game_console has 72 filled places.
Attribute laptop has 70 filled places.
Attribute pc has 70 filled places.
Attribute router has 69 filled places.
Attribute set_top_box has 70 filled places.
Attribute tablet has 70 filled places.
Attri